src/app/profile/payment-methods/payment-method/payment-method.component.js

```diff
@@ -0,0 +1,85 @@
+import angular from 'angular';
+import template from './payment-method.tpl';
+import displayAddressComponent from 'common/components/display-address/display-address.component';
+import recurringGiftsComponent from '../recurring-gifts/recurring-gifts.component';
+import paymentMethodFormModal from 'common/components/paymentMethods/paymentMethodForm/paymentMethodForm.modal.component';
+import deletePaymentMethodModal from 'common/components/paymentMethods/deletePaymentMethod/deletePaymentMethod.modal.component.js';
+import giveModalWindowTemplate from 'common/templates/giveModalWindow.tpl';
+
+class PaymentMethodController{
+
+  /* @ngInject */
+  constructor(envService,$uibModal){
+    this.isCollapsed = true;
+    this.$uibModal = $uibModal;
+    this.imgDomain = envService.read('imgDomain');
+    this.submissionError = {error: ''};
+  }
+
+  getExpiration(){
+    return `${this.model['expiry-month']}/${this.model['expiry-year']}`;
+  }
+
+  isCard(){
+    return this.model['card-number'] ? true : false;
+  }
+
+  editPaymentMethod() {
+    this.editPaymentMethodModal = this.$uibModal.open({
+      component: 'paymentMethodFormModal',
+      windowTemplateUrl: giveModalWindowTemplate.name,
+      resolve: {
+        paymentMethod: () => this.model,
+        mailingAddress: () => {}, // TODO: load this from donor details
+        onSubmit: () => this.onSubmit,
```

Comment:
You should be able to call https://github.com/CruGlobal/give-web/blob/update-payment-methods/src/common/services/api/profile.service.js#L162 to update it. An example is here https://github.com/CruGlobal/give-web/blob/update-payment-methods/src/app/profile/yourGiving/editRecurringGifts/step0/addUpdatePaymentMethod.component.js#L38